#!/usr/bin/env python3
PACKAGE = "hw08_lastname"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

follow_enum = gen.enum([ gen.const("zig_zag",  int_t, 0, "Zig-Zag"),
                         gen.const("binning",  int_t, 1, "Binning"),
                         gen.const("p_control",int_t, 2, "P Control") ],
                       "An enum to set line following type")


gen.add('enable',        bool_t,   0, 'Enable',        False)
gen.add('gain',        double_t,   0, 'Gain',               1,   0,   10)
gen.add('speed',       double_t,   0, 'Speed',              0,   0,    5)
gen.add('frame_skip',     int_t,   0, 'Frame Skip',         4,   1,   10)
gen.add("follow_type", int_t, 0,
        "Line follow algorithm", 1, 0, 2, edit_method=follow_enum)
gen.add('hsv_hue_high',     int_t,   0, 'Hue_High',            180,      0,      180)
gen.add('hsv_sat_high',     int_t,   0, 'Saturation_High',     255,      0,      255)
gen.add('hsv_val_high',     int_t,   0, 'Value_High',          255,      0,      255)

gen.add('hsv_hue_low',     int_t,   0, 'Hue_Low',            79,      0,      180)
gen.add('hsv_sat_low',     int_t,   0, 'Saturation_Low',     112,      0,      255)
gen.add('hsv_val_low',     int_t,   0, 'Value_Low',          112,      0,      255)

#gen.add('thres_value',     int_t,   0, 'Threshold_value',          0,      0,      255)


# Add hue, sat, and val low and high

exit(gen.generate(PACKAGE, "hw08", "LineFollowDynCfg"))

